### PR TITLE
[#401] Implement input bar, write & send message functionality

### LINKED ISSUE #401: Input bar + Write message
Create input bar and be able to write a message and show in the conversation.

### PR DESCRIPTION
**Pull request by ogbeche77**
Fixes #401

### CODE REVIEW

File: frontend/demo/src/actions/messages/index.ts [hunk 1]
```diff
@@ -38,6 +44,24 @@ export function listMessages(conversationId: string) {
...
+        dispatch(
+          sendMessagesAction({
+            conversationId: messagePayload.conversationId,
+            message: response,
+          })
+        );
+        return Promise.resolve(true);
+      })
+      .catch((error: Error) => {
+        return Promise.reject(error);
```
Comment (bitboxer) [inline frontend/demo/src/actions/messages/index.ts:60]: this is unneeded and the catch can be gone. The promise does this reject thing automatically

File: frontend/demo/src/pages/Inbox/MessageInput/index.tsx [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+        placeholder="Enter a message..."
+        autoFocus={true}
+        value={input}
+        onChange={handleChange}
+      />
+
+      <div className={styles.buttonSubmit}>
+        <button
+          type="button"
+          onClick={(e: React.MouseEvent<HTMLButtonElement, MouseEvent>) => {
```
Comment (bitboxer) [inline frontend/demo/src/pages/Inbox/MessageInput/index.tsx:54]: maybe move this into a method?

File: frontend/demo/src/reducers/data/messages/index.ts [hunk 3]
```diff
@@ -51,6 +50,18 @@ export default function messagesReducer(state = initialState, action: Action): a
...
       }
 
+    case getType(actions.sendMessagesAction):
+      return {
+        ...state,
+        all: {
+          ...state.all,
+          [action.payload.conversationId]: {
+            ...state.all[action.payload.conversationId],
+            ...organiseMessages([action.payload.message]),
```
Comment (bitboxer) [inline frontend/demo/src/reducers/data/messages/index.ts:60]: Not sure, but this kind looks like all the old messages in the conversation are overridden with the one from the message payload?

**Review by bitboxer — COMMENTED** [review]
Looks really nice already but I had a couple of questions. So after we cleared them I think this is ready to be merged.

File: frontend/demo/src/pages/Inbox/MessageInput/index.tsx (hunk 2 above)
Comment (AitorAlgorta) [reply frontend/demo/src/pages/Inbox/MessageInput/index.tsx:54]: Yeah, I agree. Let's move this into a method

File: frontend/demo/src/pages/Inbox/MessageInput/index.tsx [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+  useEffect(() => {
+    textAreaAdjust.current.style.height = '0px';
+    const scrollHeight = textAreaAdjust.current.scrollHeight;
+    textAreaAdjust.current.style.height = scrollHeight + 'px';
+  }, [input]);
+
+  const conversationIdParams = useParams();
+  const currentConversationId: string = conversationIdParams[Object.keys(conversationIdParams)[0]];
+
+  function handleClick() {
```
Comment (AitorAlgorta) [inline frontend/demo/src/pages/Inbox/MessageInput/index.tsx:39]: const handleClick = () =>
```suggestion
  function handleClick() {
```

**Review by AitorAlgorta — APPROVED** [review]

**Pull request merged**